lib/iris/analysis/__init__.py

```diff
@@ -1193,7 +1193,10 @@ def _proportion(array, function, axis, **kwargs):
     else:
         total_non_masked = array.shape[axis]
 
-    return _count(array, function, axis=axis, **kwargs) / total_non_masked
+    numerator = _count(array, function, axis=axis, **kwargs)
+    result = ma.asarray(numerator / total_non_masked)
```

Comment:
Bit of a hum-dinger, this ...

Using `ma.asarray(...)` to sanitise the result of the operation, so that the resulting masked array is well formed.

If this isn't done, then the `tests/unit/analysis/test_PROPORTION.Test_masked.test_ma` fails as `numerator / total_non_masked` results in a scalar masked array with a `float` value (unmasked) but an `integer` `fill_value` ... which upsets the `DataManager.data` getter.

Glad to add a comment ...